### PR TITLE
Add http_proxy configuration with unittest

### PR DESCRIPTION
**Pull request by pokoli**
This PR extends #8 by adding some unit test for the PoolManager and the proxy manager.

### CODE REVIEW

**Comment by mbohlool** [comment]
@pokoli Sorry that I missed this. Please ping me if I missed a PR like this ever in future. It should not take me more than a couple of days to do a code review. More than that is not usual :)

**Comment by mbohlool** [comment]
Please resolve the conflicts and ping me after.

File: rest_test.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unittest
+import urllib3
+
+from mock import patch
+
+from .configuration import configuration
```
Comment (mbohlool) [inline rest_test.py]: Get this from kubernetes package like if you installed the package. CI tests expect that.

Comment (pokoli) [inline rest_test.py]: Done, thanks for the hint.

**Comment by pokoli** [comment]
Running on `env TOXENV=py27 ./run_tox.sh tox` works as expected, but this si failing on travis. 

@mbohlool Any idea?

**Comment by mbohlool** [comment]
`from kubernetes.client import configuration`

`base` folder is not a proper python package.

**Comment by pokoli** [comment]
Uou, so much days without kubernetes-client lost my memory :( @mbohlool  Thanks for the tips

Now only functional tests are failing, but I dont' think this is related to my change.

**Comment by mbohlool** [comment]
Closing and reopening to see if it helps with ci failure.

File: rest_test.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    def test_poolmanager(self):
+        'Test that a poolmanager is created for rest client'
+        with patch.object(urllib3, 'PoolManager') as pool:
+            RESTClientObject(config=configuration)
+            pool.assert_called_once()
+
+    def test_proxy(self):
+        'Test that proxy is created when the config especifies it'
+        configuration.http_proxy_url = 'http://proxy.example.com'
```
Comment (mbohlool) [inline rest_test.py]: I think this is the problem, you are changing configuration singleton that has effect on other clients. You should create your own version of configuration:
```python
from kubernetes.client import Configuration
config=Configuration()
...
```

Comment (pokoli) [inline rest_test.py]: That makes sense, but it does not fix the problem. 

Indeed, running py27-functional on my laptop passes.

Comment (mbohlool) [inline rest_test.py]: The way e2e tests are setup here is not the best. They will pass if there is no cluster to run on. You need to make sure you have a minikube running on default port before run functional tests. We need to fix that but let see what is the problem of this PR.

Comment (mbohlool) [inline rest_test.py]: So the problem is still Configuration(). It should be ConfigurationObject(). Configuration() is there for backward compatibility with generated client and still returns the singleton. using ConfigurationObject() should fix things.

File: rest_test.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+from kubernetes.client import Configuration
+from kubernetes.client.rest import RESTClientObject
+
+
+class RestTest(unittest.TestCase):
+
+    def test_poolmanager(self):
+        'Test that a poolmanager is created for rest client'
+        with patch.object(urllib3, 'PoolManager') as pool:
+            RESTClientObject(config=Configuration())
```
Comment (mbohlool) [inline rest_test.py]: do you mean to pass a new config or singleton config here? `Configuration()` is the singleton. `ConfigurationObject()` will be a new one.

File: rest_test.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            pool.assert_called_once()
+
+    def test_proxy(self):
+        'Test that proxy is created when the config especifies it'
+        config = Configuration()
```
Comment (mbohlool) [inline rest_test.py]: `config = ConfigurationObject()`

**Comment by mbohlool** [comment]
I pushed a change to use ConfigurationObject and merged this because I want it to be in the release.

**Comment by pokoli** [comment]
@mbohlool thank you for your help and for finishing it!

**Pull request merged**